trino-s3-proxy/src/test/java/io/trino/s3/proxy/server/credentials/TestStsController.java

```diff
@@ -0,0 +1,80 @@
+import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.ListBucketsResponse;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+
+import java.time.Duration;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@TrinoS3ProxyTest
+public class TestStsController
+{
+    private static final String ARN = "test-arn";
+
+    private final TestingStsController stsController;
+    private final TestingS3ClientProvider s3ClientProvider;
+    private final Credential containerCredential;
+
+    @Inject
+    public TestStsController(@ForS3MockContainer Credential containerCredential, TestingStsController stsController, TestingS3ClientProvider s3ClientProvider)
+    {
+        this.containerCredential = requireNonNull(containerCredential, "containerCredential is null");
+        this.stsController = requireNonNull(stsController, "stsController is null");
+        this.s3ClientProvider = requireNonNull(s3ClientProvider, "s3ClientProvider is null");
+    }
+
+    @Test
+    public void testStsSession()
+    {
+        stsController.addAssumableRole(s3ClientProvider.credentials().emulated().accessKey(), ARN, new Credentials(new Credential("boo", "baa"), Optional.of(containerCredential)), Duration.ofHours(1));
+        try {
+            StsAssumedRole stsAssumedRole = stsController.assumeRole(new SigningMetadata(SigningService.S3, s3ClientProvider.credentials(), Optional.empty(), "us-east-1"), ARN, Optional.empty(), Optional.empty(), Optional.empty())
+                            .orElseThrow(() -> new RuntimeException("Failed to assume role"));
+
+            try (S3Client client = s3ClientProvider.newClientBuilder()
+                    .credentialsProvider(() -> AwsSessionCredentials.create(stsAssumedRole.credentials().emulated().accessKey(), stsAssumedRole.credentials().emulated().secretKey(), stsAssumedRole.session()))
+                    .build()) {
+                // valid assumed role session - should work
+                ListBucketsResponse listBucketsResponse = client.listBuckets();
+                assertThat(listBucketsResponse.buckets()).isEmpty();
+
+                stsController.removeAssumableRole(s3ClientProvider.credentials().emulated().accessKey(), ARN);
+
+                // invalid assumed role session - should fail
+                assertThatThrownBy(client::listBuckets)
+                        .asInstanceOf(InstanceOfAssertFactories.type(S3Exception.class))
+                        .extracting(S3Exception::statusCode)
+                        .isEqualTo(401);
+            }
+        }
+        finally {
+            stsController.removeAssumableRole(s3ClientProvider.credentials().emulated().accessKey(), ARN);
```

Comment:
could this be a `@AfterEach`?

Reply:
I added a `reset()` method to the test instance and now call that via `@AfterEach`